warn/documentation_test.go

```diff
@@ -0,0 +1,37 @@
+package warn
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/buildtools/testutils"
+)
+
+func TestWarningsDocumentation(t *testing.T) {
+	files, chdir := testutils.FindTests(t, "", "WARNINGS.md")
+	defer chdir()
+
+	if len(files) != 1 {
+		t.Fatalf("Expected to find exactly one WARNINGS.md file, got %d instead", len(files))
+	}
+	data, err := ioutil.ReadFile(files[0])
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	contents := string(data)
+	for _, warning := range AllWarnings {
+		link := fmt.Sprintf("  * [%s](#%s)", warning, warning)
+		if !strings.Contains(contents, link) {
+			t.Errorf("No link found for the warning %q in WARNINGS.md, is it documented?", warning)
```

Comment:
nit. the error message should include the pattern you match (`link`).

Reply:
Done.